osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -4,98 +4,281 @@
+ *
+ * E.g. Given a map with Columns 0 0 1 2 1 0
+ * 1) Note that the first note is excluded!
+ * 2) f   denotes a strain decay, then strain add operation given the current note.
+ *        f is different between global and column strain.
+ * 3) (+) denotes an add operation.
+ *
+ *                       t=0           t=1           t=2           t=3           t=4           t=5
+ * ┌───────────────────┬─────┐       ┌─────┐       ┌─────┐       ┌─────┐       ┌─────┐       ┌─────┐
+ * │ Global Strain     │  .  ├───f───►  .  ├───f───►  .  ├───f───►  .  ├───f───►  .  ├───f───►  .  │
+ * └───────────────────┴─────┘       └──┬──┘       └──┬──┘       └──┬──┘       └──┬──┘       └──┬──┘
+ *                                      │             │             │             │             │
+ *                                     (+)           (+)           (+)           (+)           (+)
+ *                                      │             │             │             │             │
+ * ┌───────────────────┬─────┐       ┌──▼──┐          │             │             │          ┌──▼──┐
+ * │ Column Strain 0   │  .  ├───f───►  .  ├──────────┼─────────────┼─────────────┼──────f───►  .  ├───►
+ * ├───────────────────┴─────┤       └──┬──┘          │             │             │          └──┬──┘
+ * ├───────────────────┬─────┤          │          ┌──▼──┐          │          ┌──▼──┐          │
+ * │ Column Strain 1   │  .  ├──────────┼──────f───►  .  ├──────────┼──────f───►  .  ├──────────┼──────►
+ * ├───────────────────┴─────┤          │          └──┬──┘          │          └──┬──┘          │
+ * ├───────────────────┬─────┤          │             │          ┌──▼──┐          │             │
+ * │ Column Strain 2   │  .  ├──────────┼─────────────┼──────f───►  .  ├──────────┼─────────────┼──────►
+ * └───────────────────┴─────┘          │             │          └──┬──┘          │             │
+ *                                      │             │             │             │             │
+ * ┌─────────────────────────┐       ┌──▼──┐       ┌──▼──┐       ┌──▼──┐       ┌──▼──┐       ┌──▼──┐
+ * │ Final Strain            │       │  .  │       │  .  │       │  .  │       │  .  │       │  .  │
+ * └─────────────────────────┘       └─────┘       └─────┘       └─────┘       └─────┘       └─────┘
+ *
+ * A few things to notice.
+ * 1) Final Strains are independent of each other.
+ * 2) Global Strain updates on every note, while Column Strain updates only for their respective columns
+ * 3) Final Strain is the summation of global and column strain.
+ */
+
 namespace osu.Game.Rulesets.Mania.Difficulty.Skills
 {
     public class Strain : StrainDecaySkill
     {
-        private const double individual_decay_base = 0.125;
-        private const double overall_decay_base = 0.30;
+        /// <summary> Individual Strain Decay Exponent Base. Used in <see cref="applyDecay"/> </summary>
+        private const double decay_base = 0.125;
```

Comment:
This sort of xmldoc really just takes up an extra line of source. If you renamed this to `individual_strain_decay_base` you would achieve the same effect, more or less.

Reply:
Changed it to `column_decay_base` and removed xml as it's self-doced